File: python/881. Boats to Save People.py
class Solution(object):
    def numRescueBoats(self, people, limit):
        """
        :type people: List[int]
        :type limit: int
        :rtype: int
        """
        ans = 0
        n = len(people)
        l, r = 0, n-1
        people.sort()
        while l <= r:
            two = people[l] + people[r]
            if two <= limit:
                l += 1
                r -= 1
            else:
                r -= 1
            ans += 1
        return ans
        

File: go/src/leetcode/881. Boats to Save People/881. Boats to Save People.go
package main

import (
	"fmt"
	"sort"
)

func numRescueBoats(people []int, limit int) int {
	var ships int
	sort.Ints(people)
	l, r := 0, len(people) - 1
	for l <= r {
		w := people[l] + people[r]
		if w <= limit {
			l++
		}
		r--
		ships++
	}
	fmt.Println(ships)
	return ships
}

func main() {
	numRescueBoats([]int{1}, 3)
	numRescueBoats([]int{1,2}, 3)
	numRescueBoats([]int{3,2,2,1}, 3)
	numRescueBoats([]int{3,5,3,4}, 5)
	numRescueBoats([]int{5,1,4,2}, 6)
}
